### ISSUE TITLE
[@aws-cdk/aws-events] Schedule cannot be created from CfnParameter

### ISSUE THREAD

**Issue by alopix** [aws/aws-cdk#9413]
When trying to use a parameter for the duration of a events Schedule, on synth, an error occurs: `"Unable to perform time unit conversion on un-resolved token -1.8881545897087682e+289."`

Specifying the duration for e.g. a Lambda timeout works, but not if the Duration is used in a `Schedule.rate`.

### Reproduction Steps

```
import { Stack, App, Duration, CfnParameter } from '@aws-cdk/core';
import { Rule, Schedule } from '@aws-cdk/aws-events';

const invokeScheduleInMinutes = new CfnParameter(this, 'invokeScheduleInMinutes', {
      type: 'Number',
      description: 'The minutes between scheduled lambda invocations.'
});

new Rule(this, 'Rule', {
      ruleName: `cron`,
      schedule: Schedule.rate(Duration.minutes(invokeScheduleInMinutes.valueAsNumber))
});
```

### Error Log
```
…/node_modules/@aws-cdk/core/lib/duration.ts:289
    throw new Error(`Unable to perform time unit conversion on un-resolved token ${amount}.`);
          ^
Error: Unable to perform time unit conversion on un-resolved token -1.8881545897087682e+289.
    at convert (…/node_modules/@aws-cdk/core/lib/duration.ts:289:11)
    at Duration.toSeconds (…node_modules/@aws-cdk/core/lib/duration.ts:127:12)
    at Function.rate (…/node_modules/@aws-cdk/aws-events/lib/schedule.ts:20:18)
    at new AppStack (…/infrastructure/stacks/src/AppStack.ts:47:26)
    at Object.<anonymous> (…/infrastructure/stacks/src/app.ts:21:1)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Module.m._compile (…/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Object.require.extensions.<computed> [as .ts] (…/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
```

### Environment

  - **CLI Version      :** 1.56.0
  - **Framework Version:** 1.56.0
  - **Node.js Version:** v12.16.1
  - **OS               :** macOS
  - **Language (Version):** TypeScript 3.9.7

### Other

--- 

This is :bug: Bug Report

**Comment by rix0rrr** [comment]
As a workaround, use `Schedule.expression()` for now.

---

For fixing this:

`Duration` should should have an `isUnresolved()` method, and `Schedule.rate` should just render the number in the given unit if the duration is unresolved.

Same for other `rate()` methods in the library.

**Comment by raphael-riel** [comment]
For the records, here is a working example for the proposed workaround:

```typescript
        const rateMinutes = new cdk.CfnParameter(this, "RateMinutes", {
            default: 5,
            description: "...",
            type: "Number",
        });

        new events.Rule(this, "ScheduledFlush", {
            schedule: events.Schedule.expression(`rate(${rateMinutes.value} minutes)`),
            targets: [
                new eventsTargets.LambdaFunction(flushFunction),
            ],
        });
```

`cron(...)` expression is also possible. Ref: https://github.com/aws-samples/aws-cdk-examples/blob/master/typescript/lambda-cron/index.ts#L22

**Comment by alopix** [comment]
Thanks for the workaround suggestions.
This does not solve the problem though. It will work for the schedule, but e.g. a Metric period is still gonna fail the same way, as it requires a resolved Duration value and does not work from a Parameter.

**Comment by jumic** [comment]
The code in this issue can be used without error in the meanwhile.
It was resolved in PR #13064. This issue is still open because it wasn't referenced correctly in the PR.

@peterwoodworth Could you please close this issue?

**Comment by peterwoodworth** [comment]
Will do, thank you
